4.5 Improve rendering of error pages

markstory opened the pull request:
PHP exceptions don't include the source line/file in the stacktrace so our error pages started on the frame *after* the line where the error actually was. This makes debugging harder as the problematic line isn't as obvious.

These changes add an unclosable frame before each stack that shows the line the exception came from. I've also removed the `activeFrame` logic as we don't want to show two adjacent frames open by default.

![Screenshot from 2023-04-09 00-15-51](https://user-images.githubusercontent.com/24086/230754393-557c78f2-6fec-402c-8535-555e63f1d39b.png)

markstory committed:
Improve rendering of error pages

PHP exceptions don't include the source line/file in the stacktrace so
our error pages started on the frame *after* the line where the error
actually was. This makes debugging harder as the problematic line isn't
as obvious.

These changes add an unclosable frame before each stack that shows the
line the exception came from.

ADmad commented:
Shouldn't we fix this in 4.x itself?

markstory commented:
> Shouldn't we fix this in 4.x itself?

We *could* but we'd still need to re-do the solution in 4.next as well. Remember that error pages look totally different in 4.5 + 5.x

ADmad commented:
> Remember that error pages look totally different in 4.5 + 5.x

Ah.. I forgot about that.

othercorey commented:
I think 4.5 is ok for this. It's a long-standing issue and users don't need the fix for stability.

LordSimal commented:
LGTM 👍🏻

LordSimal commented:
Acutally there is one thing missing in my opinion: The `edit` link which is present for all other frames bellow.

markstory committed:
Refine styling of stack frames

- Reorder file/line/function
- Add padding.
- Add edit for generated 0th frame.

markstory commented:
![Screenshot from 2023-04-11 22-39-04](https://user-images.githubusercontent.com/24086/231333776-3f63256f-74ca-4630-b4cd-134c39021018.png)

I've made a few more changes:

- Reordered` the file/line/function.
- Added some more padding in the excerpts
- Added an edit button for the top frame.
